web/src/routes/ProjectView/ProjectView.js

```diff
@@ -67,6 +70,14 @@ function ProjectViewInner({
         }, 100)
       }
   }
+  const instance = useRef()
+
```

Comment:
let's pull the value `10000` out into a `const` defined at the top of the file, with a descriptive comment above it?

Also could use a descriptive comment for this `useEffect`, since this is the beating heart of all the rest of this logic, which could be otherwise hard to grasp